ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/DataStoreModule.java

```diff
@@ -17,25 +17,116 @@
 
 import org.hyperledger.besu.ethereum.chain.BlockchainStorage;
 import org.hyperledger.besu.ethereum.core.BlockHeaderFunctions;
+import org.hyperledger.besu.ethereum.storage.keyvalue.KeyValueSegmentIdentifier;
 import org.hyperledger.besu.ethereum.storage.keyvalue.KeyValueStoragePrefixedKeyBlockchainStorage;
+import org.hyperledger.besu.plugin.services.BesuConfiguration;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
 import org.hyperledger.besu.plugin.services.storage.KeyValueStorage;
+import org.hyperledger.besu.plugin.services.storage.SegmentIdentifier;
+import org.hyperledger.besu.plugin.services.storage.rocksdb.RocksDBKeyValueStorageFactory;
+import org.hyperledger.besu.plugin.services.storage.rocksdb.RocksDBMetricsFactory;
+import org.hyperledger.besu.plugin.services.storage.rocksdb.configuration.RocksDBCLIOptions;
+import org.hyperledger.besu.services.kvstore.InMemoryKeyValueStorage;
+import org.hyperledger.besu.services.kvstore.LimitedInMemoryKeyValueStorage;
 
+import java.util.List;
+import java.util.function.Supplier;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import com.google.common.base.Suppliers;
 import dagger.Module;
 import dagger.Provides;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
-@SuppressWarnings("WeakerAccess")
+@SuppressWarnings({"CloseableProvides"})
 @Module(includes = GenesisFileModule.class)
 public class DataStoreModule {
 
+  private static final Logger LOG = LogManager.getLogger();
+  private final Supplier<RocksDBKeyValueStorageFactory> rocksDBFactory =
```

Comment:
Does this need to be a supplier? I think the db is only created when you call `create`

Reply:
It's a supplier so we don't initialize it unless we actually need it.